CefSharp/Internals/JavascriptCallback.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright © 2010-2014 The CefSharp Project. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+using System;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+
+namespace CefSharp.Internals
+{
+    [DataContract]
+    internal sealed class JavascriptCallback : IJavascriptCallback
+    {
+        private readonly long id;
+        private readonly int browserId;
+        private readonly WeakReference _browserProcess;
+        private bool disposed;
+
+        public JavascriptCallback(long id, int browserId, BrowserProcessServiceHost browserProcess)
+        {
+            this.id = id;
+            this.browserId = browserId;
+            _browserProcess = new WeakReference(browserProcess);
+        }
+
+        private BrowserProcessServiceHost BrowserProcessServiceHost
+        {
+            get { return (BrowserProcessServiceHost)_browserProcess.Target; }
```

Comment:
What happens when `Target = null`? Prob better to use `as`. Also no real need for the property, just cast using `as` inline.